builder/triton/driver_triton.go

```diff
@@ -0,0 +1,167 @@
+package triton
+
+import (
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/joyent/gosdc/cloudapi"
+	"github.com/mitchellh/packer/packer"
+)
+
+type driverTriton struct {
+	client *cloudapi.Client
+	ui     packer.Ui
+}
+
+func NewDriverTriton(ui packer.Ui, config Config) (Driver, error) {
+	client, err := config.AccessConfig.CreateTritonClient()
+	if err != nil {
+		return nil, err
+	}
+
+	return &driverTriton{
+		client: client,
+		ui:     ui,
+	}, nil
+}
+
+func (d *driverTriton) CreateImageFromMachine(machineId string, config Config) (string, error) {
+	opts := cloudapi.CreateImageFromMachineOpts{
+		Machine:     machineId,
+		Name:        config.ImageName,
+		Version:     config.ImageVersion,
+		Description: config.ImageDescription,
+		Homepage:    config.ImageHomepage,
+		EULA:        config.ImageEULA,
+		ACL:         config.ImageACL,
+		Tags:        config.ImageTags,
+	}
+
+	image, err := d.client.CreateImageFromMachine(opts)
+	if err != nil {
+		return "", err
+	}
+
+	return image.Id, err
+}
+
+func (d *driverTriton) CreateMachine(config Config) (string, error) {
+	opts := cloudapi.CreateMachineOpts{
+		Package:         config.MachinePackage,
+		Image:           config.MachineImage,
+		Networks:        config.MachineNetworks,
+		Metadata:        config.MachineMetadata,
+		Tags:            config.MachineTags,
+		FirewallEnabled: config.MachineFirewallEnabled,
+	}
+
+	if config.MachineName != "" {
+		opts.Name = config.MachineName
+	}
```

Comment:
Yeah that does make more sense. I've modified the code and doc to make the builder use `packer-builder-[image_name]`. I've not added the version because in theory the version string is allowed to contain characters which a VM name isn't.